internal/archiver/archiver.go

```diff
@@ -798,7 +798,13 @@ func (arch *Archiver) Snapshot(ctx context.Context, targets []string, opts Snaps
 
 		return arch.saveTree(wctx, tree)
 	}()
-	debug.Log("saved tree, error: %v", err)
+
+	if err != nil {
```

Comment:
I think the proper fix would be to change line 806 from
```
if err == nil || errors.Cause(err) == context.Canceled {
```
to
```
if err == nil || (errors.Cause(err) == context.Canceled && werr != nil) {
```

If err is reports a canceled context, then the snapshot creation has definitely failed. However, werr might contain more of an explanation so we should return that.